src/macrofier/engine.mjs

```diff
@@ -934,12 +934,30 @@ function generateExamples(json = {}, mainTemplates = {}, languages = {}) {
   return examples
 }
 
-function generateMethods(json = {}, examples = {}, templates = {}) {
+function generateMethods(json = {}, examples = {}, templates = {}, options = { polymorphic: false }) {
```

Comment:
I think we should keep this function simple: it takes a list of method schemas and outputs the method bodies.

Let's move the code to insert additional methods higher up into generateMacros, right after we call `promoteAndNameSubSchemas`.

Reply:
Ok, will move

Reply:
moved generateMacros after handling promoteAndNameSubSchemas